### PR TITLE
Issue 451: Support multiple embedded react native controls in UWP

### PR DESCRIPTION
**Pull request by nmyhre**
To support multiple react-native-windows in a XAML UserControl in a future commit we will need to support non-singleton patterns. UWP does not allow more than one progress dialog so we need to handle that case when more than one react-native-windows control is present and re-loading.

### CODE REVIEW

File: ReactWindows/ReactNative.Shared/DevSupport/DevSupportManager.cs [hunk 1]
```diff
@@ -346,21 +346,36 @@ public Task<bool> IsPackagerRunningAsync()
...
+#if WINDOWS_UWP
+        static ProgressDialog _currentprogressDialog = null; // UWP can only have one
+#endif
+
         public async void HandleReloadJavaScript()
         {
             DispatcherHelpers.AssertOnDispatcher();
 
             HideRedboxDialog();
-            HideDevOptionsDialog();
```
Comment (rozele) [inline ReactWindows/ReactNative.Shared/DevSupport/DevSupportManager.cs]: What's the reason for no longer dismissing the dev options dialog?

**Comment by rozele** [comment]
I get the point of this, which is to prevent multiple progress dialogs from popping up simultaneously, but is it really the complete approach here? Assuming the goal is to support multiple RN views loading simultaneously, shouldn't this run the loading dialog until all operations are complete, rather than only tracking a single loading operation?

**Comment by nmyhre** [comment]
I agree and would classify this as a workaround and not a complete solution, indeed. It's possible as well that the correct solution for embedded applications is to suppress the dialog completely or at least give that option. In embedded applications it is hard to know if there is a dialog already up for another reason and we will conflict now with the host. Also, not knowing what demand there is for a packaging like this I am unsure how to resolve the tension with respect to investing in an area that no one is currently using, yet. What would you think about simply preventing the dialog from appearing always and then see if the issue comes up in usage? As I am new to this project I will defer to your guidance here.

File: ReactWindows/ReactNative.Shared/DevSupport/DevSupportManager.cs (hunk 1 above)
Comment (nmyhre) [inline ReactWindows/ReactNative.Shared/DevSupport/DevSupportManager.cs]: That must be a merge issue? I did not intend to remove that at all. I will check into what happened there. Thanks.

**Comment by rozele** [comment]
@nmyhre I think this whole component could use re-architecting. Rather than having a bunch of different functions for hiding dialogs and expecting to make sure we hide all the right dialogs at the right time, we really need a singleton dialog manager that handles use cases like "hide current dialog and show this one" and "continue showing current dialog but wire up additional cancellation token". 

That's a much larger task to take on, so I'll file an issue with that recommendation, and I think your recommendation of simply having an option to not show any progress dialogs is fine.

**Comment by rozele** [comment]
Opened #1273 in reference to above.

**Comment by nmyhre** [comment]
@rozele, I modified the code to make the progress dialog optional in general. Seems like it might be useful maybe?

File: ReactWindows/ReactNative.Shared/DevSupport/DevSupportManager.cs [hunk 2]
```diff
@@ -353,39 +359,54 @@ public Task<bool> IsPackagerRunningAsync()
...
+#endif
+                token = progressDialog.Token;
             }
             else
             {
-                progressDialog.Topmost = true;
-                progressDialog.WindowStartupLocation = WindowStartupLocation.CenterScreen;
+                // Progress not enabled - provide empty implementations
+                cancel = () => { };
+                token = default(CancellationToken);
```
Comment (rozele) [inline ReactWindows/ReactNative.Shared/DevSupport/DevSupportManager.cs]: >default(CancellationToken); [](start = 24, length = 27)

Not sure if there's any difference between default(CancellationToken) and CancellationToken.None. Presumably not.

**Pull request merged**